opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -613,8 +619,205 @@ class SimulatorFullyImplicitBlackoilEbos
+               wtotals[1] += wr[0][w];
+               //Field Gas produced
+               wtotals[2] += wr[2][w];
+           }
+           if (is_injector){
+               //Field Oil injected
+               wtotals[6] += wr[1][w];
+               //Field Water injected
+               wtotals[7] += wr[0][w];
+               //Field Gas injected
+               wtotals[8] += wr[2][w];
+           }
+        }
+        for (int w = 0; w < nw; ++w) {
+           const bool is_producer = wells->type[w] == PRODUCER;
+           if (is_producer) {
+               if (pu.phase_used[Water] && pu.phase_used[Oil]) {
+                   //Field Watercut
+                   wtotals[3] = wtotals[1]/(wtotals[0] + wtotals[1] + wtotals[2]);
+               }
+               if (pu.phase_used[Oil] && pu.phase_used[Gas]) {
+                   //Field GOR
+                   wtotals[4] = wtotals[2]/wtotals[0];
+               }
+               if (pu.phase_used[Water] && pu.phase_used[Gas]) {
+                   //Field WGR       
+                   wtotals[5] = wtotals[1]/wtotals[2];
+               }
+	   }
+        }
+        return wtotals;
+   }
+   
+   //Output well report to the .PRT file
+   void outputWellReport(std::vector<std::vector<double>> wr,
+             std::vector<double> wtotals,
+             const Wells* wells,
+             WellState& xw,
+             const UnitSystem& units)
+    {           
+        int nw = wells->number_of_wells;
+        const std::string mode[4] = { "BHP", "THP", "RESERVOIR_RATE", "SURFACE_RATE" };
+        std::ostringstream pr;
+        pr << "\n                                                       PRODUCTION REPORT \n"
+           << "                                                       ................. \n\n"
+           << "----------------------------------------------------------------------------------------------------------------------------- \n"
+           << ":       WELL       :     CTRL     :     OIL     :    WATER    :     GAS     : WATER : GAS/OIL: WAT/GAS:  BHP OR  :  THP OR  : \n"
+           << ":       NAME       :     MODE     :     RATE    :    RATE     :     RATE    :  CUT  :  RATIO :  RATIO :  CON.PR. :  BLK.PR. : \n";
+        if (units.getType() == UnitSystem::UnitType::UNIT_TYPE_METRIC) {
+           pr << ":      OR GRID     :              :    SM3/DAY  :    SM3/DAY  :    SM3/DAY  :       : SM3/SM3: SM3/SM3:   BARSA  :   BARSA  : \n";
+        }
+        if (units.getType() == UnitSystem::UnitType::UNIT_TYPE_FIELD) {
+           pr << ":      OR GRID     :              :    STB/DAY  :    STB/DAY  :   MSCF/DAY  :       :MSCF/STB:STB/MSCF:   PSIA   :   PSIA   : \n";
+        }
+           pr << "============================================================================================================================= \n"
+              << ":                  :              :             :             :             :       :        :        :          :          :\n"
```

Comment:
I think this line only makes the reports more space-consuming without making them more readable. Also, it is harder to exctract/copy/paste the output to tools such as octave for quick inspection.

Preferably, the colons separating the columns should go as well for the same reason (as long as whitespace is sufficient to clearly separate columns).